spec/models/vacancy_spec.rb

```diff
@@ -673,4 +673,71 @@
       end
     end
   end
+
+  describe "duplicate and conflict validations" do
+    subject(:vacancy) { build(:vacancy, :external, attributes) }
+
+    let(:school) { create(:school) }
+    let(:publisher_ats_api_client) { create(:publisher_ats_api_client) }
+
+    let(:attributes) do
+      {
+        job_title: "Math Teacher",
+        expires_at: "2025-01-01",
+        organisations: [school],
+        publisher_ats_api_client: publisher_ats_api_client,
+        external_reference: "REF123",
+        job_advert: "A job advert",
+        external_advert_url: "https://example.com",
+        phases: %w[primary],
+      }
+    end
+
+    describe "conflict validation" do
+      let(:conflict_attributes) do
+        {
+          publisher_ats_api_client: publisher_ats_api_client,
+          external_reference: "REF123",
+        }
+      end
+
+      it "is invalid when a vacancy with same ATS client ID and external reference exists" do
+        create(:vacancy, :external, conflict_attributes)
+
+        expect(vacancy).not_to be_valid
+        expect(vacancy.errors[:base])
+          .to include("A vacancy with the provided ATS client ID and external reference already exists.")
+      end
+
+      it "stores the conflicting vacancy" do
+        conflicted_vacancy = create(:vacancy, :external, conflict_attributes)
+
+        expect(vacancy.find_conflicting_vacancy).to eq(conflicted_vacancy)
+      end
+    end
+
+    describe "duplicate validation" do
+      let(:duplicate_attributes) do
+        {
+          job_title: "Math Teacher",
+          expires_at: "2025-01-01",
+          organisations: [school],
+        }
+      end
+
+      it "is invalid when a vacancy with same job_title, expires_at and organisations exists" do
+        create(:vacancy, :external, duplicate_attributes)
+
+        expect(vacancy).not_to be_valid
+        expect(vacancy.errors[:base])
+          .to include("A vacancy with the same job title, expiry date, and organisation already exists.")
+      end
+
+      it "stores the conflicting vacancy" do
```

Comment:
Again, I might be misunderstanding this but I thought we _didn't_ want to store them